xbmc/addons/GUIDialogAddonInfo.cpp

```diff
@@ -336,6 +359,31 @@ void CGUIDialogAddonInfo::OnInstall()
   if (i != -1)
   {
     Close();
+    if (HELPERS::ShowYesNoDialogText(CVariant{m_item->GetAddonInfo()->Name()},
+                                     CVariant{39020}) == DialogResponse::YES)
+    {
+      std::map<std::string, std::string> added;
+
+      if (m_item->GetAddonInfo()->NoDeps(added) > 0)
+      {
+        std::string show="1";
+        std::string id = m_item->GetAddonInfo()->ID();
+        std::string name = m_item->GetAddonInfo()->Name();
+        while (!show.empty())
+        {
+          show = ShowDependencyList(id);
+          if (!show.empty())
+          {
+            AddonPtr addon;
+            if (CServiceBroker::GetAddonMgr().FindInstallableById(show,addon))
+              ShowForItem(CFileItemPtr(new CFileItem(addon)));
+          }
+        }
+        if (HELPERS::ShowYesNoDialogText(CVariant{name},
+                                         CVariant{39021}) != DialogResponse::YES)
+          return;
+      }
```

Comment:
This is way too many dialogs to click though. Just make the select dialog say "The following add-ons will be installed." Continue/Cancel, or something. No need for either of the yesno dialogs